### PR TITLE
fix: upgrade hashicorp/google to 5.0

### PR DESCRIPTION
**Pull request by LloydArmstrong**
When trying to upgrade the `hashicorp/google` version to 5.0, the `scheduled_queries` submodule fails due to it's requirement of `~>4.0`. 

My familiarity with this specific submodule is not extensive and therefor, I'm not entirely sure of breaking changes. Please let me know if this is not support as of yet.

### CODE REVIEW

**Comment by LloydArmstrong** [comment]
I realise this is a much bigger issue due to the dependancy on other Terraform Modules that have the same constraints, (even just within the testing module)

- kms
- project-factory

Even just to test this, it will require parallel MR's with the same versioning changes. It will require a concerted and synchronised effort between all modules involved

File: test/setup/versions.tf [hunk 1]
```diff
@@ -19,11 +19,11 @@ terraform {
   required_providers {
     google = {
       source  = "hashicorp/google"
-      version = ">= 3.53.0, < 5.0"
+      version = ">= 3.53.0, ~> 5.0"
```
Comment (apeabody) [inline test/setup/versions.tf:22]:
```suggestion
      version = ">= 3.53.0, < 6"
```

File: test/setup/versions.tf [hunk 2]
```diff
@@ -19,11 +19,11 @@ terraform {
...
     }
     google-beta = {
       source  = "hashicorp/google-beta"
-      version = ">= 3.53.0, < 5.0"
+      version = ">= 3.53.0, ~> 5.0"
```
Comment (apeabody) [inline test/setup/versions.tf:26]:
```suggestion
      version = ">= 3.53.0, < 6"
```

File: versions.tf [hunk 3]
```diff
@@ -20,7 +20,7 @@ terraform {
 
     google = {
       source  = "hashicorp/google"
-      version = ">= 4.42, < 5.0"
+      version = ">= 4.42, ~> 5.0"
```
Comment (apeabody) [inline versions.tf:23]:
```suggestion
      version = ">= 4.42, < 6"
```

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks @LloydArmstrong! - We should avoid making this a breaking change by leaving the minimum version requirements intact and lifting the maximum.

**Comment by LloydArmstrong** [comment]
I've fixed the commit message convention used. Please take another look. Thanks.

**Review by apeabody — APPROVED** [review]

**Pull request merged**